front/src/applications/stdcm/utils/formatStdcmConfV2.ts

```diff
@@ -113,82 +115,112 @@ export const checkStdcmConf = (
       })
     );
   }
+
   const startTime = dateTimeToIso(`${originDate}T${originTime}`);
-  if (!startTime) {
-    error = true;
-    dispatch(
-      setFailure({
-        name: t('operationalStudies/manageTrainSchedule:errorMessages.trainScheduleTitle'),
-        message: t('operationalStudies/manageTrainSchedule:errorMessages.noOriginTime'),
-      })
-    );
-  } else if (searchDatetimeWindow) {
-    const startDatetime = new Date(startTime);
-    if (startDatetime < searchDatetimeWindow.begin || searchDatetimeWindow.end < startDatetime) {
+
+  const startDateTime = (() => {
```

Comment:
It's a bit weird to instantiate this function in order to call it directly after. You don't need this function, use a `let startDatetime: Date | undefined` and add the code below :)